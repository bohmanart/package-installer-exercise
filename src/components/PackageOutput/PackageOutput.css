.PackageOutput {
    background: #f6f6f6;
    border: 1px solid #d6d6d6;
    border-radius: .5rem;
    color: #dd2154;
    padding: 1rem;
    white-space: normal;
}
