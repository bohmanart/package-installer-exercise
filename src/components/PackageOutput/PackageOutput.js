import React from 'react'

const PackageOutput = ({output}) => (
    <div>
        <h2>Output</h2>
        <samp>"{output}"</samp>
    </div>
)

export default PackageOutput
