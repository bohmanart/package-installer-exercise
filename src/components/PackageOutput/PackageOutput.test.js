import React from 'react'
import renderer from 'react-test-renderer'
import PackageOutput from './PackageOutput'

describe('<PackageOutput/>', () => {
    
    it('should render `h2` and `samp` elements', () => {
        const tree = renderer.create(
            <PackageOutput/>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('should render a sample output of a comma seperated list of package names from a string passed in', () => {
        const tree = renderer.create(
            <PackageOutput output="KittenService, Ice, Cyberportal, Leetmeme, CamelCaser, Fraudstream" />
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

})

