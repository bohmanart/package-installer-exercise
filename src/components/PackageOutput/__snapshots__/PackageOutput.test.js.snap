// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageOutput/> should render \`h2\` and \`samp\` elements 1`] = `
<div>
  <h2>
    Output
  </h2>
  <samp>
    "
    "
  </samp>
</div>
`;

exports[`<PackageOutput/> should render a sample output of a comma seperated list of package names from a string passed in 1`] = `
<div>
  <h2>
    Output
  </h2>
  <samp>
    "
    KittenService, Ice, Cyberportal, Leetmeme, CamelCaser, Fraudstream
    "
  </samp>
</div>
`;
